modules/pokemon.py

```diff
@@ -719,6 +723,10 @@ class Pokemon:
 
     def __init__(self, data: bytes):
         self.data = data
+        if "Unown" in self.species.name and self.species.name != f"Unown ({self.unown_letter})":
```

Comment:
Note: `self.species.name != f"Unown ({self.unown_letter})"` is required as "Unown" species' general name is overwritten by this code, this means if you encounter a "Z" then a "!", the "!" would have come through as "Z" otherwise.